vllm/distributed/parallel_state.py

```diff
@@ -962,19 +967,31 @@ def ensure_kv_transfer_initialized(vllm_config: "VllmConfig") -> None:
     Initialize KV cache transfer parallel group.
     """
 
-    global _KV_TRANSFER
+    global _KV_CONNECTOR_AGENT
 
     if vllm_config.kv_transfer_config is None:
         return
 
     if all([
             vllm_config.kv_transfer_config.is_kv_transfer_instance,
-            _KV_TRANSFER is None
+            _KV_CONNECTOR_AGENT is None
     ]):
-        _KV_TRANSFER = kv_transfer.KVTransferAgent(
-            rank=get_world_group().rank,
-            local_rank=get_world_group().local_rank,
-            config=vllm_config)
+        from vllm.distributed.kv_transfer.kv_connector.factory import (
+            KVConnectorFactory)
+        from vllm.distributed.kv_transfer.kv_connector.v1 import (
+            KVConnectorRole as KVConnectorRole_V1)
+
+        kwargs = {
+            "rank": get_world_group().rank,
+            "local_rank": get_world_group().local_rank,
+            "config": vllm_config,
+        }
+        if envs.VLLM_USE_V1:
+            # NOTE(Kuntai):
+            # Parallel state is initialized in v1 worker,
+            # so this connector is for sure worker connector.
+            kwargs["role"] = KVConnectorRole_V1.WORKER
```

Comment:
```suggestion
            "config": vllm_config,
            # NOTE(Kuntai):
            # Parallel state is initialized in v1 worker,
            # so this connector is for sure worker connector.
            "role": KVConnectorRole_V1.WORKER
        }
```

This KVConnectorRole_V1 is necessary to call KVConnectorFactory.create_connector just later